File: samples/FBLoginSample/src/main/java/com/facebook/fbloginsample/requests/PermissionRequest.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.fbloginsample.requests;

import com.facebook.AccessToken;
import com.facebook.GraphRequest;
import com.facebook.HttpMethod;

public class PermissionRequest {
  private static final String PERMISSIONS_ENDPOINT = "/me/permissions";
  private static final String APP = "app";

  public static void makeRevokePermRequest(String permission, GraphRequest.Callback callback) {
    String graphPath;
    if (permission.equals(APP)) {
      graphPath = PERMISSIONS_ENDPOINT;
    } else {
      graphPath = PERMISSIONS_ENDPOINT + "/" + permission;
    }

    GraphRequest request =
        GraphRequest.newGraphPathRequest(AccessToken.getCurrentAccessToken(), graphPath, callback);
    request.setHttpMethod(HttpMethod.DELETE);
    request.executeAsync();
  }
}

File: samples/FBLoginSample/src/main/java/com/facebook/fbloginsample/requests/PermissionRequest.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.fbloginsample.requests

import com.facebook.AccessToken.Companion.getCurrentAccessToken
import com.facebook.GraphRequest
import com.facebook.GraphRequest.Companion.newGraphPathRequest
import com.facebook.HttpMethod

object PermissionRequest {
  private const val PERMISSIONS_ENDPOINT = "/me/permissions"
  private const val APP = "app"

  @JvmStatic
  fun makeRevokePermRequest(permission: String, callback: GraphRequest.Callback?) {
    val graphPath =
        if (permission == APP) {
          PERMISSIONS_ENDPOINT
        } else {
          "$PERMISSIONS_ENDPOINT/$permission"
        }
    val request = newGraphPathRequest(getCurrentAccessToken(), graphPath, callback)
    request.httpMethod = HttpMethod.DELETE
    request.executeAsync()
  }
}
